gadget/install/install.go

```diff
@@ -771,6 +773,11 @@ func FactoryReset(model gadget.Model, gadgetRoot string, kernelSnapInfo *KernelS
 		// device) for each role
 		deviceForRole[vs.Role] = onDiskStruct.Node
 
+		// TODO: when save partition does not have slots, then
+		// we need to create new partition the old fashion
+		// way.  Or, we should upgrade the save partition
+		// (which should be safe since the seed should not be
+		// considered safe to downgrade).
```

Comment:
I think I need to create a spread test for that case first.
 1. Install with an old snapd.
 2. Remodel with a newer snapd.
 3. Factory reset with the new snapd.
It is possible that it is already working, but we are left with some non-slot keys that we should remove.